service/src/main/kotlin/io/provenance/digitalcurrency/consortium/frameworks/BurnCompletionQueue.kt

```diff
@@ -0,0 +1,93 @@
+package io.provenance.digitalcurrency.consortium.frameworks
+
+import io.provenance.digitalcurrency.consortium.api.DepositFiatRequest
+import io.provenance.digitalcurrency.consortium.bankclient.BankClient
+import io.provenance.digitalcurrency.consortium.config.CoroutineProperties
+import io.provenance.digitalcurrency.consortium.config.logger
+import io.provenance.digitalcurrency.consortium.config.withMdc
+import io.provenance.digitalcurrency.consortium.domain.CoinBurnRecord
+import io.provenance.digitalcurrency.consortium.domain.TxStatus
+import io.provenance.digitalcurrency.consortium.extension.isFailed
+import io.provenance.digitalcurrency.consortium.extension.mdc
+import io.provenance.digitalcurrency.consortium.service.PbcService
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.springframework.context.event.EventListener
+import org.springframework.stereotype.Component
+import java.util.UUID
+
+class CoinBurnDirective(
+    override val id: UUID
+) : Directive()
+
+class CoinBurnOutcome(
+    override val id: UUID
+) : Outcome()
+
+@Component
+class CoinBurnQueue(
+    coroutineProperties: CoroutineProperties,
+    private val pbcService: PbcService,
+    private val bankClient: BankClient
+) :
+    ActorModel<CoinBurnDirective, CoinBurnOutcome> {
+    private val log = logger()
+
+    @EventListener(DataSourceConnectedEvent::class)
+    fun startProcessing() {
+        log.info("start burn queueing framework")
+        start()
+    }
+
+    override val numWorkers: Int = coroutineProperties.numWorkers
+    override val pollingDelayMillis: Long = coroutineProperties.pollingDelayMs
+
+    override suspend fun loadMessages(): List<CoinBurnDirective> =
+        transaction {
+            CoinBurnRecord.findPending().map { CoinBurnDirective(it.id.value) }
+        }
+
+    override fun processMessage(message: CoinBurnDirective): CoinBurnOutcome {
+        transaction {
+            CoinBurnRecord.findPendingForUpdate(message.id).first().let { coinBurn ->
+                withMdc(*coinBurn.mdc()) {
+                    val response = pbcService.getTransaction(coinBurn.txHash!!)!!.txResponse
+
+                    when (response == null || response.isFailed()) {
+                        true -> {
+                            log.info("burn failed, need to retry")
+                            coinBurn.resetForRetry()
+                        }
+                        false -> {
+                            if (coinBurn.coinRedemption != null) {
```

Comment:
if we decouple setting the status to COMPLETE via event stream and actionable followup like the deposit fiat request as a separate column (or another status), we wouldnt have to do the getTransaction call above

Reply:
But wouldn't we still need/want to do a getTransaction in the event stream to make sure it didn't fail?

Reply:
could be wrong but I thought event stream only included tx's in blocks that were successful? weird if it includes failed tx's...